webarena/03_docker_create_containers.sh

```diff
@@ -36,16 +36,27 @@ cp ../openstreetmap-templates/docker-compose.yml ./docker-compose.yml
 cp ../openstreetmap-templates/leaflet.osm.js ./vendor/assets/leaflet/leaflet.osm.js
 cp ../openstreetmap-templates/fossgis_osrm.js ./app/assets/javascripts/index/directions/fossgis_osrm.js
 
+# sed works differently on Mac (BSD) and Linux (GNU),
+# so we need to check the version of sed to determine the correct syntax for in-place editing
+if [[ -z "$OSTYPE" ]]; then
+  echo "Error: OSTYPE is not set. Please run this script in a proper shell environment."
+  exit 1
+fi
+if [[ "$OSTYPE" == "darwin"* ]]; then
+  SED_INPLACE=(-i '')  # MacOS
+else
+  SED_INPLACE=(-i)  # Linux
+fi
 # set up web server port
-sed -i "s|MAP_PORT|${MAP_PORT}|g" docker-compose.yml
+sed "${SED_INPLACE[@]}" "s|MAP_PORT|3000|g" docker-compose.yml
```

Comment:
Is this one intended?
```suggestion
sed "${SED_INPLACE[@]}" "s|MAP_PORT|${MAP_PORT}|g" docker-compose.yml
```

Reply:
Good catch! I've hardcoded this on my local setup, but should obviously not do so here.